superset/datasets/commands/create.py

```diff
@@ -0,0 +1,82 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from flask_appbuilder.security.sqla.models import User
+from marshmallow import UnmarshalResult, ValidationError
+
+from superset.commands.base import BaseCommand, CommandValidateReturn
+from superset.datasets.commands.base import populate_owners
+from superset.datasets.commands.exceptions import (
+    DatabaseNotFoundValidationError,
+    DatasetCreateFailedError,
+    DatasetExistsValidationError,
+    DatasetInvalidError,
+    TableNotFoundValidationError,
+)
+from superset.datasets.dao import DatasetDAO
+
+
+class CreateDatasetCommand(BaseCommand):
+    def __init__(self, user: User, unmarshal: UnmarshalResult):
+        self._actor = user
+        self._properties = unmarshal.data.copy()
+        self._errors = unmarshal.errors
+
+    def run(self):
+        is_valid, exceptions = self.validate()
+        if not is_valid:
+            for exception in exceptions:
+                self._errors.update(exception.normalized_messages())
+            raise DatasetInvalidError()
+
+        dataset = DatasetDAO.create(self._properties)
+
+        if not dataset:
+            raise DatasetCreateFailedError()
+        return dataset
+
+    def validate(self) -> CommandValidateReturn:
+        is_valid, exceptions = True, list()
```

Comment:
do we need `is_valid` here? Can't we just rely on the emptiness of the `exceptions` list to imply success?